### PR TITLE
ci(docs): add `make docs/output` to copy all raw docs

### PR DESCRIPTION
**Pull request by lahabana**
The goal of this is to take the configuration files (crds, proto json schema, yaml confs) with no modification and get them in a directory.

These will be then used in the docs website where they can be modified as needed. That way the logic stays in one place (the website).

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Comment by lahabana** [comment]
This will be backported and we'll then have a new job that opens a PR with all these folders.
The hope is to have as little logic as possible here and have everything in the respective docs repo

File: mk/docs.mk [hunk 1]
```diff
@@ -32,3 +33,19 @@ docs/generated/kuma-cp.md: ## Generate Mesh API reference
...
+	cp $(DOCS_CP_CONFIG) $(DOCS_OUTPUT_DIR)/kuma-cp.yaml
+	cp $(HELM_VALUES_FILE) $(DOCS_OUTPUT_DIR)/helm-values.yaml
+	mkdir $(DOCS_OUTPUT_DIR)/crds
+	for f in $$(find deployments/charts -name '*.yaml' | grep '/crds/'); do cp $$f $(DOCS_OUTPUT_DIR)/crds/; done
+
+	mkdir -p $(DOCS_OUTPUT_DIR)/proto
+	$(PROTOC) \
+		--jsonschema_out=$(DOCS_OUTPUT_DIR)/proto \
+		--plugin=protoc-gen-jsonschema=$(PROTOC_GEN_JSONSCHEMA) \
+		$(DOCS_PROTOS)
```
Comment (bartsmykla) [inline mk/docs.mk:51]:
```suggestion

.PHONY: docs/output
docs/output: clean/docs/output | $(DOCS_OUTPUT_DIR)
	cp $(DOCS_CP_CONFIG) $(DOCS_OUTPUT_DIR)/kuma-cp.yaml
	cp $(HELM_VALUES_FILE) $(DOCS_OUTPUT_DIR)/helm-values.yaml

	for f in $$(find deployments/charts -name '*.yaml' | grep '/crds/'); do cp $$f $(DOCS_OUTPUT_DIR)/crds/; done

	$(PROTOC) \
		--jsonschema_out=$(DOCS_OUTPUT_DIR)/proto \
		--plugin=protoc-gen-jsonschema=$(PROTOC_GEN_JSONSCHEMA) \
		$(DOCS_PROTOS)

.PHONY: clean/docs/output
clean/docs/output:
	rm -rf $(DOCS_OUTPUT_DIR)

$(DOCS_OUTPUT_DIR):
	mkdir -p $@/{crds,proto}
```

To confirm everything works the same after my above suggestion:
```sh
…/kuma (updateDocs (6128136)) [!] ❯ make docs/output
rm -rf /kuma/build/docs
mkdir -p /kuma/build/docs/{crds,proto}
cp pkg/config/app/kuma-cp/kuma-cp.defaults.yaml /kuma/build/docs/kuma-cp.yaml
cp "deployments/charts/kuma/values.yaml" /kuma/build/docs/helm-values.yaml
for f in $(find deployments/charts -name '*.yaml' | grep '/crds/'); do cp $f /kuma/build/docs/crds/; done
/Users/bs/.kuma-dev/kuma-release-2.3/bin/protoc --proto_path=/Users/bs/.kuma-dev/kuma-release-2.3/protos --proto_path=. --proto_path=. \
                --jsonschema_out=/kuma/build/docs/proto \
                --plugin=protoc-gen-jsonschema=/Users/bs/.kuma-dev/kuma-release-2.3/bin/protoc-gen-jsonschema \
                api/mesh/v1alpha1/*.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=Selector.json msg_name=Selector proto_filename=selector.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=CircuitBreaker.json msg_name=CircuitBreaker proto_filename=circuit_breaker.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=Metrics.json msg_name=Metrics proto_filename=metrics.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=MetricsBackend.json msg_name=MetricsBackend proto_filename=metrics.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=PrometheusMetricsBackendConfig.json msg_name=PrometheusMetricsBackendConfig proto_filename=metrics.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=PrometheusAggregateMetricsConfig.json msg_name=PrometheusAggregateMetricsConfig proto_filename=metrics.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=PrometheusEnvoyConfig.json msg_name=PrometheusEnvoyConfig proto_filename=metrics.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=EnvoyAdmin.json msg_name=EnvoyAdmin proto_filename=envoy_admin.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=Dataplane.json msg_name=Dataplane proto_filename=dataplane.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=DataplaneInsight.json msg_name=DataplaneInsight proto_filename=dataplane_insight.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=DiscoverySubscription.json msg_name=DiscoverySubscription proto_filename=dataplane_insight.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=DiscoverySubscriptionStatus.json msg_name=DiscoverySubscriptionStatus proto_filename=dataplane_insight.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=DiscoveryServiceStats.json msg_name=DiscoveryServiceStats proto_filename=dataplane_insight.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=Version.json msg_name=Version proto_filename=dataplane_insight.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=KumaDpVersion.json msg_name=KumaDpVersion proto_filename=dataplane_insight.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=EnvoyVersion.json msg_name=EnvoyVersion proto_filename=dataplane_insight.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=DataplaneOverview.json msg_name=DataplaneOverview proto_filename=dataplane_overview.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=KumaResource.json msg_name=KumaResource proto_filename=kds.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=XDSConfigRequest.json msg_name=XDSConfigRequest proto_filename=kds.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=XDSConfigResponse.json msg_name=XDSConfigResponse proto_filename=kds.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=StatsRequest.json msg_name=StatsRequest proto_filename=kds.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=StatsResponse.json msg_name=StatsResponse proto_filename=kds.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=ClustersRequest.json msg_name=ClustersRequest proto_filename=kds.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=ClustersResponse.json msg_name=ClustersResponse proto_filename=kds.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=ExternalService.json msg_name=ExternalService proto_filename=externalservice.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=FaultInjection.json msg_name=FaultInjection proto_filename=fault_injection.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=MeshGateway.json msg_name=MeshGateway proto_filename=gateway.proto
INFO[0000] Generating JSON-schema for stand-alone ENUM   enum_name=HttpMethod jsonschema_filename=HttpMethod.json proto_filename=http_method.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=MeshGatewayRoute.json msg_name=MeshGatewayRoute proto_filename=gateway_route.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=HealthCheck.json msg_name=HealthCheck proto_filename=health_check.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=Mesh.json msg_name=Mesh proto_filename=mesh.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=CertificateAuthorityBackend.json msg_name=CertificateAuthorityBackend proto_filename=mesh.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=Networking.json msg_name=Networking proto_filename=mesh.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=Tracing.json msg_name=Tracing proto_filename=mesh.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=TracingBackend.json msg_name=TracingBackend proto_filename=mesh.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=DatadogTracingBackendConfig.json msg_name=DatadogTracingBackendConfig proto_filename=mesh.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=ZipkinTracingBackendConfig.json msg_name=ZipkinTracingBackendConfig proto_filename=mesh.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=Logging.json msg_name=Logging proto_filename=mesh.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=LoggingBackend.json msg_name=LoggingBackend proto_filename=mesh.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=FileLoggingBackendConfig.json msg_name=FileLoggingBackendConfig proto_filename=mesh.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=TcpLoggingBackendConfig.json msg_name=TcpLoggingBackendConfig proto_filename=mesh.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=Routing.json msg_name=Routing proto_filename=mesh.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=MeshInsight.json msg_name=MeshInsight proto_filename=mesh_insight.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=Message.json msg_name=Message proto_filename=mux.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=ProxyTemplate.json msg_name=ProxyTemplate proto_filename=proxy_template.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=ProxyTemplateSource.json msg_name=ProxyTemplateSource proto_filename=proxy_template.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=ProxyTemplateProfileSource.json msg_name=ProxyTemplateProfileSource proto_filename=proxy_template.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=ProxyTemplateRawSource.json msg_name=ProxyTemplateRawSource proto_filename=proxy_template.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=ProxyTemplateRawResource.json msg_name=ProxyTemplateRawResource proto_filename=proxy_template.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=RateLimit.json msg_name=RateLimit proto_filename=rate_limit.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=Retry.json msg_name=Retry proto_filename=retry.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=ServiceInsight.json msg_name=ServiceInsight proto_filename=service_insight.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=Timeout.json msg_name=Timeout proto_filename=timeout.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=TrafficLog.json msg_name=TrafficLog proto_filename=traffic_log.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=TrafficPermission.json msg_name=TrafficPermission proto_filename=traffic_permission.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=TrafficRoute.json msg_name=TrafficRoute proto_filename=traffic_route.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=TrafficTrace.json msg_name=TrafficTrace proto_filename=traffic_trace.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=VirtualOutbound.json msg_name=VirtualOutbound proto_filename=virtual_outbound.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=ZoneIngress.json msg_name=ZoneIngress proto_filename=zone_ingress.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=ZoneIngressInsight.json msg_name=ZoneIngressInsight proto_filename=zone_ingress_insight.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=ZoneEgress.json msg_name=ZoneEgress proto_filename=zoneegress.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=ZoneEgressInsight.json msg_name=ZoneEgressInsight proto_filename=zoneegressinsight.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=ZoneEgressOverview.json msg_name=ZoneEgressOverview proto_filename=zoneegressoverview.proto
INFO[0000] Generating JSON-schema for MESSAGE            jsonschema_filename=ZoneIngressOverview.json msg_name=ZoneIngressOverview proto_filename=zoneingress_overview.proto
```

**Review by bartsmykla — APPROVED** [review]
One suggestion, but LGTM overall

**Review by bartsmykla — APPROVED** [review]

File: mk/docs.mk [hunk 2]
```diff
@@ -51,4 +53,4 @@ clean/docs/output:
 	rm -rf $(DOCS_OUTPUT_DIR)
 
 $(DOCS_OUTPUT_DIR):
-	mkdir -p $@/{crds,proto}
+	mkdir -p $@
```
Comment (bartsmykla) [inline mk/docs.mk:56]: It seemed to me as a problem with different shell on CI, so this should be sufficient also:
```sh
	mkdir -p $@/crds $@/proto
```

Comment (lahabana) [reply mk/docs.mk:56]: Yes I split them too as it makes sense for the base dir to be created but the direct dir should be likely created in the same place. It's easier to read

**Pull request merged**